### PR TITLE
Fix/gp/model

### PR DESCRIPTION
**Pull request by Chlenix**

### CODE REVIEW

File: src/main/java/no/fdk/dataservicecatalog/config/SecurityConfig.java [hunk 1]
```diff
@@ -46,6 +47,7 @@ CorsConfigurationSource corsConfiguration() {
     @Bean
     SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
         return http
+//                .cors().configurationSource(corsConfiguration()).and()
```
Comment (azaxarov) [inline src/main/java/no/fdk/dataservicecatalog/config/SecurityConfig.java:50]: intentional?

File: src/main/java/no/fdk/dataservicecatalog/dto/acat/TermsAndConditions.java [hunk 2]
```diff
@@ -4,17 +4,19 @@
...
+    private String price;
 
-    private String usageLimitation;
+    private Map<String, String> usageLimitation;
 
-    private String performance;
+    private String capacityAndPerformance;
 
-    private String availability;
+    private String reliability;
```
Comment (azaxarov) [inline src/main/java/no/fdk/dataservicecatalog/dto/acat/TermsAndConditions.java:20]: If I'm not mistaken, all fields in the class should be translatable strings, i.e. Map<String, String>

File: src/main/java/no/fdk/dataservicecatalog/repository/DataServiceMongoRepository.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import reactor.core.publisher.Mono;
 
 public interface DataServiceMongoRepository extends ReactiveMongoRepository<DataService, String> {
-    Mono<DataService> findByIdAndCatalogId(String dataServiceId, String catalogId);
-    Mono<Long> deleteByIdAndCatalogId(String dataServiceId, String catalogId);
-    Flux<DataService> findAllByCatalogId(String catalogId);
+    Mono<DataService> findByIdAndOrganizationId(String dataServiceId, String catalogId);
+    Mono<Long> deleteByIdAndOrganizationId(String dataServiceId, String catalogId);
+    Flux<DataService> findAllByOrganizationId(String catalogId);
```
Comment (azaxarov) [inline src/main/java/no/fdk/dataservicecatalog/repository/DataServiceMongoRepository.java:11]: All occurrences of catalogId can be renamed to organizationId

File: src/main/java/no/fdk/dataservicecatalog/service/DataServiceService.java [hunk 4]
```diff
@@ -69,27 +60,27 @@ private DataService applyPatch(DataService dataService, Map<String, Object> part
                 .doOnError(error -> log.error("dataservice with id {} failed mapping {}", dataServiceId, error.getMessage()));
         return dataServiceMono.flatMap(dataService -> {
             dataService.setId(dataServiceId);
-            dataService.setCatalogId(catalogId);
+            dataService.setOrganizationId(catalogId);
```
Comment (azaxarov) [inline src/main/java/no/fdk/dataservicecatalog/service/DataServiceService.java:63]: rename catalogId to organizationId

File: src/test/java/no/fdk/dataservicecatalog/utils/TestData.java [hunk 5]
```diff
@@ -159,7 +159,7 @@
                 DataService
                         .builder()
                         .id(String.format("%s/id-10", catalogId))
-                        .catalogId(catalogId)
+                        .organizationId(catalogId)
```
Comment (azaxarov) [inline src/test/java/no/fdk/dataservicecatalog/utils/TestData.java:162]: rename catalogId

File: src/main/java/no/fdk/dataservicecatalog/security/PermissionManager.java [hunk 6]
```diff
@@ -35,18 +37,18 @@ public static PermissionManager of(String resourceId, String permission) {
...
     public Mono<AuthorizationDecision> check(Mono<Authentication> mono, AuthorizationContext authorizationContext) {
         return mono.cast(OpenIDAuthenticationToken.class).map(auth -> {
             if (auth.isAuthenticated() && auth.getStatus().equals(OpenIDAuthenticationStatus.SUCCESS)) {
-                var catalogId = extractCatalogIdFromPath(authorizationContext.getExchange().getRequest().getPath());
-                var allowed = getResourceRoles(auth.getAuthorities()).stream().anyMatch(rr -> rr.matchPermission(resourceId, catalogId, permission));
-                return new AuthorizationDecision(allowed);
+                var path = authorizationContext.getExchange().getRequest().getPath().pathWithinApplication().value();
+                if (path.startsWith("/catalogs/")) {
+                    Matcher matcher = catalogIdPattern.matcher(path);
+                    if (matcher.find()) {
```
Comment (azaxarov) [inline src/main/java/no/fdk/dataservicecatalog/security/PermissionManager.java:47]: What is the purpose of this check? Don't we already have something similar configured at a higher level?

File: src/main/java/no/fdk/dataservicecatalog/repository/DataServiceMongoRepository.java (hunk 3 above)
Comment (Chlenix) [reply src/main/java/no/fdk/dataservicecatalog/repository/DataServiceMongoRepository.java:11]: it's called catalogId in url, but organizationId in json body.

File: src/main/java/no/fdk/dataservicecatalog/security/PermissionManager.java (hunk 6 above)
Comment (Chlenix) [reply src/main/java/no/fdk/dataservicecatalog/security/PermissionManager.java:47]: We make sure that the path is actually `/catalogs/{id}/...` and not something like `/some/future/path/with/catalogs/{id}/...`

File: src/main/java/no/fdk/dataservicecatalog/service/DataServiceService.java (hunk 4 above)
Comment (Chlenix) [reply src/main/java/no/fdk/dataservicecatalog/service/DataServiceService.java:63]: catalogId means it comes from the url path

**Pull request merged**